Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -507,6 +507,15 @@ RenderLayer* RenderObject::enclosingLayer() const
     return nullptr;
 }
 
+RenderLayer* RenderObject::enclosingCompositedLayer() const
```

Comment:
I'd still prefer that you don't add this. Talk to RenderLayer, and have it find the enclosing composited layer.